apps/meteor/app/discussion/server/hooks/propagateDiscussionMetadata.ts

```diff
@@ -62,7 +61,7 @@ callbacks.add(
 callbacks.add(
 	'afterDeleteRoom',
 	async (rid) => {
-		RoomsSync.find({ prid: rid }, { fields: { _id: 1 } }).forEach(({ _id }: Pick<IRoom, '_id'>) => deleteRoom(_id));
+		await Rooms.find({ prid: rid }, { projection: { _id: 1 } }).forEach(({ _id }: Pick<IRoom, '_id'>) => deleteRoom(_id));
```

Comment:
using `await ... forEach()` will not await for all async operations on the `forEach` callbacks..

since this is a heavy operation, I'd change it to a `for await` in this case.. wdyt?

Reply:
Ah, that's because deleteRoom is not async yet 👀 was planning to remove it with the Promise.awaits. Can change this to be for/await/of tho